File: api/src/handlers/starter_templates/jog_contribute.c
/* Reference Jar-of-Greed bot in C. Contributes a fixed amount each round.
 * Wire protocol: judge/protocols/wire.md (stdio transport).
 * Game spec: judge/protocols/jar-of-greed.md. */
#include <stdio.h>
#include <string.h>

#define CONTRIBUTION 1

int main(void) {
    char line[4096];
    setvbuf(stdout, NULL, _IOLBF, 0);
    while (fgets(line, sizeof line, stdin)) {
        char ev[16], kind[32];
        long seq;
        if (sscanf(line, "%15s %ld %31s", ev, &seq, kind) < 3) continue;
        if (strcmp(ev, "EVENT") != 0) continue;
        if (!strcmp(kind, "GAME_STARTED") || !strcmp(kind, "ROUND_RESULT")) {
            printf("ACT CONTRIBUTE %d\n", CONTRIBUTION);
        } else if (!strcmp(kind, "GAME_END")) {
            return 0;
        }
    }
    return 0;
}

File: api/src/handlers/starter_templates/jog_contribute.rs
// Reference Jar-of-Greed bot. Contributes a fixed amount each round.
//
// Wire protocol: judge/protocols/wire.md (stdio transport).
// Game spec: judge/protocols/jar-of-greed.md.
//
// Strategy: on GAME_STARTED contribute, then re-contribute each time
// a ROUND_RESULT signals the next round has begun. The room rejects
// duplicate CONTRIBUTEs in the same round, so one ACT per round is
// the safe pattern.

use std::io::{self, BufRead, Write};

const CONTRIBUTION: u32 = 1;

fn main() {
    let stdin = io::stdin();
    let mut stdout = io::stdout();
    for line in stdin.lock().lines() {
        let line = match line {
            Ok(l) => l,
            Err(_) => break,
        };
        let mut tok = line.split_whitespace();
        if tok.next() != Some("EVENT") {
            continue;
        }
        let _seq = tok.next();
        match tok.next() {
            Some("GAME_STARTED") | Some("ROUND_RESULT") => {
                let _ = writeln!(stdout, "ACT CONTRIBUTE {CONTRIBUTION}");
                let _ = stdout.flush();
            }
            Some("GAME_END") => break,
            _ => {}
        }
    }
}
